### PR TITLE
Implement analyze command with package tuneup

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

File: lib/src/analyze_command.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import 'dart:async';
 import 'dart:io';
 
 import 'package:args/command_runner.dart';
+import 'package:path/path.dart' as p;
```
Comment (devoncarew) [inline lib/src/analyze_command.dart]: I generally prefix as `path`, but YMMV.

File: lib/src/analyze_command.dart [hunk 2]
```diff
@@ -22,8 +23,54 @@ class AnalyzeCommand extends Command<Null> {
...
+          await runAndStream('flutter', <String>['packages', 'get'], package);
+      if (exitCode != 0) {
+        print(
+            'ERROR: Unable to run "flutter packages get" in package $package.');
+        throw new ToolExit(1);
+      }
+    }
+
+    final List<String> failingPackages = <String>[];
+    await for (Directory package in _listAllPluginPackages(packagesDir)) {
```
Comment (devoncarew) [inline lib/src/analyze_command.dart]: This works. You could also try running tuneup once on the `packages/` dir, which should end up with the same result. I don't think there's a big advantage either way.

File: lib/src/analyze_command.dart (hunk 1 above)
Comment (goderbauer) [inline lib/src/analyze_command.dart]: The other commands in this package use `p` as well, so I am going with consistency here :)

File: lib/src/analyze_command.dart (hunk 2 above)
Comment (devoncarew) [inline lib/src/analyze_command.dart]: Ah, you're collecting the list of failed packages; nevermind.

Comment (goderbauer) [inline lib/src/analyze_command.dart]: tuneup doesn't like that:

```
~/dev/plugins/packages (master) $ pub global run tuneup check
FileSystemException: Cannot open file, path = '/Users/goderbauer/dev/plugins/packages/pubspec.yaml' (OS Error: No such file or directory, errno = 2)
#0      _File.throwIfError (dart:io/file_impl.dart:599)
#1      _File.openSync (dart:io/file_impl.dart:454)
#2      _File.readAsBytesSync (dart:io/file_impl.dart:514)
#3      _File.readAsStringSync (dart:io/file_impl.dart:548)
#4      Project.pubspec (package:tuneup/src/common.dart:103)
#5      Project.name (package:tuneup/src/common.dart:86)
#6      CheckCommand.execute (package:tuneup/commands/check.dart:25)
<asynchronous suspension>
#7      TuneupCommand.run (package:tuneup/src/common.dart:33)
#8      CommandRunner.runCommand (package:args/command_runner.dart:194)
<asynchronous suspension>
#9      Tuneup.run (package:tuneup/tuneup.dart:90)
<asynchronous suspension>
#10     main (http://localhost:64656/tuneup.dart:13)
#11     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:263)
#12     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)
```

**Comment by devoncarew** [comment]
lgtm!

File: lib/src/analyze_command.dart (hunk 2 above)
Comment (devoncarew) [inline lib/src/analyze_command.dart]: Ah, yeah, the analysis server doesn't like directories that are names `packages/` - you are better off analyzing each dir individually.

Comment (devoncarew) [inline lib/src/analyze_command.dart]: (though, tuneup shouldn't throw here)

Comment (devoncarew) [inline lib/src/analyze_command.dart]: https://github.com/google/tuneup.dart/issues/63

**Comment by devoncarew** [comment]
Note that tuneup will use the analysis server from the sdk that it is running from. If analyzing a flutter project, it would be best to use the analysis server from the dart-sdk in the futter sdk. I assume the dart on the path on travis is the one from the flutter sdk?

**Comment by goderbauer** [comment]
Yes, your assumption s correct.

**Pull request merged**